test/abstractarray.jl

```diff
@@ -709,3 +709,16 @@ let
     @test !issparse(m1)
     @test !issparse(m2)
 end
+
+#isinteger and isreal
+@test isinteger(Diagonal(rand(1:5,5)))
+@test isreal(Diagonal(rand(5)))
```

Comment:
If I understand what you're asking, no. I think if it's an `AbstractArray` of `Any`s then this calls `all(isinteger())` on each element.